spine-c/src/spine/SkeletonJson.c

```diff
@@ -299,75 +300,74 @@ static spAnimation* _spSkeletonJson_readAnimation (spSkeletonJson* self, Json* r
 		}
 		for (frame = constraintMap->child, i = 0; frame; frame = frame->next, ++i) {
 			spTransformConstraintTimeline_setFrame(timeline, i, Json_getFloat(frame, "time", 0), Json_getFloat(frame, "rotateMix", 1),
-												   Json_getFloat(frame, "translateMix", 1), Json_getFloat(frame, "scaleMix", 1), Json_getFloat(frame, "shearMix", 1));
-			readCurve(SUPER(timeline), i, frame);
+					Json_getFloat(frame, "translateMix", 1), Json_getFloat(frame, "scaleMix", 1), Json_getFloat(frame, "shearMix", 1));
+			readCurve(frame, SUPER(timeline), i);
 		}
 		animation->timelines[animation->timelinesCount++] = SUPER_CAST(spTimeline, timeline);
-		duration = timeline->frames[constraintMap->size * 5 - 5];
+		duration = timeline->frames[constraintMap->size * 3 - 3];
```

Comment:
spine-libgdx is using the "entries" timeline constants here and in all the similar places in SkeletonJson and SkeletonBinary. Eg `TRANSFORMCONSTRAINT_ENTRIES`. Those constants would need to move to a header (Animation.h or extension.h? probably Animation.h I guess).